Выполнил домашнее задание №6

MAMKUH6OPIIIUK opened the pull request:
Переписал приложение на ORM.
Добавил таблицу `comments`, реализовал crud для неё. Двусторонних связей с книгами не делал. Сделал только ссылку на книгу в комментарии (как рекомендовали на занятии).
N+1 решал без графов - только с `@Fetch` и `@BatchSize`, а в методе findAll репозитория книг добавляю авторов сразу через `join fetch`.

В процессе переписывания откатил модели обратно к сущностям, содержащим объекты.

MAMKUH6OPIIIUK committed:
Выполнил домашнее задание №6

vlad-iv commented on spring-hw-06/src/main/java/ru/otus/spring/homework/oke/exceptions/AuthorBooksFoundException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.otus.spring.homework.oke.exceptions;
+
+public class AuthorBooksFoundException extends RuntimeException {
+    public AuthorBooksFoundException(String message) {
```
Можно выбрасывать NotFoundException , так как обработка исключений одинаковая

vlad-iv commented on spring-hw-06/src/main/java/ru/otus/spring/homework/oke/repositories/JpaBooksRepository.java:
```diff
@@ -0,0 +1,56 @@
+
+@Repository
+@RequiredArgsConstructor
+public class JpaBooksRepository implements BooksRepository {
```
Предлагаю в названиях классов использовать единственное число
BooksRepository  -> BookRepository 
Так же в названии пакетов тоже, можно единственное число
models -> model

vlad-iv commented on spring-hw-06/src/main/java/ru/otus/spring/homework/oke/services/AuthorsServiceImpl.java:
```diff
@@ -0,0 +1,64 @@
+    @Override
+    @Transactional(readOnly = true)
+    public AuthorResponseDto findById(Long id) {
+        Author author = this.authorsRepository.findById(id).orElse(null);
```
Предлагаю использовать подход с orElseThrow (вместо orElse)

MAMKUH6OPIIIUK replied:
Спасибо! Переделал 👍 Так получилось лаконичнее

MAMKUH6OPIIIUK committed:
Опциональные правки по результатам проверки ДЗ№6:
- переименованы package и классы (теперь всё в единственном числе)
- проверка отсутствия сущностей в БД и проброс исключений теперь сделаны через .orElseThrow, а не через проверку на null
- убрал все исключения, кроме NotFoundException
